lib/participant.js

```diff
@@ -348,14 +340,14 @@ class Participant extends EventEmitter {
         ? new RemoteTrack(sid, trackTransceiver, options)
         : new RemoteTrack(sid, trackTransceiver, isEnabled, isSwitchedOff, switchOffReason, setPriority, setRenderHint, options);
 
-      self._addTrack(track, publication, trackTransceiver.id);
+      self._addTrack(track, publication);
```

Comment:
This is confusing. `_addTrack` method on this class takes `track` and `key` as parameters, but looks like derived class takes `track` and `publication` as parameter's.  
By invoking `self._addTrack` here with `publication` as parameter means this class knows about the derived class. I suggest we add different method to distinguish these two.
`_addTrackWithPublication`, and `_addTrackWithKey`

Reply:
@makarandp0 You're right. This is a known issue, since the actual solution is to move all this code to the RemoteParticipant class. But, I don't want to do that in this PR.

Reply:
@makarandp0 I'm going to merge this in order to cut an RC today. Thanks for the review!